packages/utils/src/ethereumHelper.ts

```diff
@@ -10,9 +10,6 @@ export async function getSignature(
   if (!ethereum.request) throw new Error('No `request` On Ethereum Provider');
 
   let params = [msg, address.toLowerCase()];
-  if (ethereum.isMetaMask) {
-    params = [params[1], params[0]];
-  }
```

Comment:
Ok, so `isMetaMask` is set for both MetaMask and Rabby, so the code in question was getting called.

Removing the reversal of the params doesn't seem to affect MetaMask's ability to connect, so I guess you were right in your assessment. How it ever worked is beyond me…

Reply:
MetaMask do compatible with both order but this will make some wired things if you set address as the first parameter(like probably not work with Trezor hardware wallet)